### PR TITLE
Using `options` to configure `rust_source`

### PR DESCRIPTION
**Pull request by Ilia-Kosenkov**
Using `options` to provide default values for `rust_source`, as discussed in #35.

### CODE REVIEW

File: R/zzz.R [hunk 1]
```diff
@@ -1,9 +1,38 @@
...
+    rextendr.patch.crates_io =  c(
+      'extendr-api = { git = "https://github.com/extendr/extendr" }',
+      'extendr-macros = { git = "https://github.com/extendr/extendr" }'
+    ),
+    # character scalar
+    # Default version of 'extendr_api' if no 'patch.crates_io' is specified
+    rextendr.extendr.version = "*",
+    # character scalar
+    # Default version of 'extendr_macros' if no 'patch.crates_io' is specified
+    rextendr.extendr_macros.version = "*"
```
Comment (clauswilke) [inline R/zzz.R:31]: I'm a bit unclear about when to use `.` vs `_` as a separator but I guess it's fine the way you set it up.

**Review by clauswilke — APPROVED** [review]

File: R/zzz.R (hunk 1 above)
Comment (Ilia-Kosenkov) [reply R/zzz.R:31]: @clauswilke,
From what I observed in options present on my system and set by other packages -- `'.'` is used as a separator, e.g. `{ggplot2}` https://github.com/tidyverse/ggplot2/blob/9741da5050f81b7b5c012c56d02f45fc93d68f89/R/scale-colour.r#L43-L45.
So the naming convention looks like `<package>.<global_option>` and `<package>.<entity>.<property>`.
Usage of `'_'` is adopted from rextendr itself, where it replaces `'-'`, e.g. `patch.crates_io` is an equivalent of Cargo.toml's `[patch.crates-io]`. Same for `extendr-*`.

**Comment by Ilia-Kosenkov** [comment]
@clauswilke,
I have added simple tests to check if `options` have any effect on `rust_source` (by supplying invalid option values and runnig `rust_function`, expecting Rust compilation error).
If there are no issues, I can merge this (after tests finish once again).

**Comment by clauswilke** [comment]
Yes, looks good.

**Comment by clauswilke** [comment]
Oh, you should also add your name to the authors list in `DESCRIPTION`, at the end.

**Comment by Ilia-Kosenkov** [comment]
@clauswilke,
I have another PR ready, I will do it in the next one, thank you.
Merging this now.

**Pull request merged**